<launch>

  <!-- usb_cam -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/videoCAM" />
    <param name="autoexposure" value="false"/>
    <param name="exposure" value="100"/>
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="pixel_format"     value="yuyv"                                         />
    <param name="camera_frame_id"  value="usb_cam"                                      />
    <param name="camera_name"      value="usb_cam"                                      />
    <param name="camera_info_url"  value="file://$(find soosang)/src/usb_cam.yaml"  />
    <param name="io_method" value="mmap"/>
  </node>

  <!-- ar tag -->
  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    <param name="marker_size"                type="double" value="9.0" />
    <param name="max_new_marker_error"       type="double" value="0.05" />
    <param name="max_track_error"            type="double" value="0.05" />
    <param name="output_frame"               type="string" value="usb_cam" />
    <remap from="camera_image" to="/usb_cam/image_raw"/>
    <remap from="camera_info" to="/usb_cam/camera_info"/>
  </node>

  <!-- lidar -->
  <node name="rplidar_node"  pkg="xycar_lidar"  type="xycar_rplidar" output="screen" respawn="false" >
    <param name="serial_port"         type="string" value="/dev/ttyLIDAR"/>
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser_frame"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
  </node>

  <!-- imu -->
  <node pkg="xycar_imu" type="9dof_imu_node.py" name="xycar_imu" output="screen">
    <param name="rviz_mode" type="string" value="false" />
  </node>

  <!-- ultrasonic -->
  <include file="$(find xycar_ultrasonic)/launch/xycar_ultrasonic.launch" />

  <node pkg="xycar_ultrasonic" type="xycar_ultrasonic_viewer.py" name="ultrasonic_viewer" output="screen" />
  
  <!-- motor -->
  <arg name="motor_type" default="$(env motor_version)" />
  <arg name="angle_offset" default="0" />
  <group unless="$(arg motor_type)">
    <include file="$(find vesc_driver)/launch/vesc_drive_xycar_motor.launch" />
  </group>
  <node name="xycar_motor" pkg="xycar_motor" type="xycar_motor.py" output="screen" >
    <param name="motor_type" value="$(arg motor_type)" />
    <param name="angle_offset" value="$(arg angle_offset)" />
  </node>

  <!-- main script -->
  <node name="auto_drive" pkg="soosang" type="main.py" output="screen"/>

</launch>
